File: app/src/javax/java/com/adawolfs/mvp/login/LoginPresenter.java
package com.adawolfs.mvp.login;

import com.adawolfs.mvp.mvp.BaseMvpPresenter;
import com.adawolfs.mvp.mvp.BaseMvpPresenterImpl;

import javax.inject.Inject;

/**
 * Created by adawolfs on 1/20/17.
 */

public class LoginPresenter extends BaseMvpPresenterImpl<LoginContract.View>
        implements LoginContract.Presenter {

    private LoginContract.Interactor mInteractor;

    public LoginPresenter(LoginContract.Interactor interactor, LoginContract.View view) {
        mInteractor = interactor;
        attachView(view);
    }

    @Override
    public void validateCredentials(String user, String password) {
        getView().showError("Error en el usuario Java");
    }

}

File: app/src/kotlin/java/com/adawolfs/mvp/login/LoginPresenter.kt
package com.adawolfs.mvp.login

import com.adawolfs.mvp.mvp.BaseMvpInteractor
import com.adawolfs.mvp.mvp.BaseMvpPresenterImpl

/**
 * Created by adawolfs on 1/20/17.
 */

                    //This is the constructor definition for this class
class LoginPresenter(interactor: LoginContract.Interactor, view: LoginContract.View) :
        BaseMvpPresenterImpl<LoginContract.View>(view), LoginContract.Presenter{

    init {
        attachView(view)
    }

    private var mInteractor : LoginContract.Interactor = interactor

    override fun validateCredentials(user: String, password: String) {
        getView()?.showError("Error en el usuario Kotlin")
    }


}